.github/workflows/smoketest.yml

```diff
@@ -0,0 +1,391 @@
+      run: |
+        rm -rf build src-cache
+        mkdir build
+        mkdir src-cache
+
+    # Subsequent cache hit builds use cache-timestamp to ensure all sources are marked 'old',
+    # except those changed between the cache SHA and the SHA being built.
+    - name: Build
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: |
+        cd build
+        if [[ ! -f cache-timestamp ]]; then touch cache-timestamp; fi
+        touch ../cmake-timestamp
+        ../cmake/cmake ../src -Wno-dev -DRUNTIME_USER=$(whoami) -DRUNTIME_GROUP=$(id -gn) -DDESTDIR=$(realpath ..)/install -DINCLUDE_PLUGINS=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_PYTHON2=0 -DUSE_PYTHON3=1 -DSUPPRESS_SPARK=1 -DUSE_CPPUNIT=1 -DUSE_CASSANDRA=Off -DCMAKE_CXX_FLAGS="-DBUILD_TAG=\\\"dummytag\\\" -DBUILD_VERSION_MAJOR=1 -DBUILD_VERSION_MINOR=1 -DBUILD_VERSION_POINT=0" -DBUILD_TAG="smoketest"
+        make -j$(nproc) install
+
+    # Cache contains:
+    # - make binaries (build-cache)
+    # - Any sources written since cache-timestamp within the source dir (src-cache) (e.g. AWS writes a config header)
+    # - cache-timestamp, timestamp of cache-miss build. All files except diffs set to this time, ensuring no existing binaries are rebuilt
+    # - cache-sha, the SHA of the commit built. On a cache-hit, used to find changed files and ensure they are only ones rebuilt.
+    - name: Prepare cache
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: |
+        echo ${{ steps.vars.outputs.sha }} > build/cache-sha
+        mv build build-cache
+        cd src
+        find -name '.git*' -prune -o -newer ../cmake-timestamp -type f -print | cpio -p -dum ../src-cache
+
+    # - alter stock ecl-test.json, to point to install dir (using jq)
+    # - create a clean copy of the install directory (for later reuse)
+    # - rm hpcc-data from copy (the post-setup hpcc-data will be moved in when done)
+    - name: Prepare install artifact
+      if: steps.cache.outputs.cache-hit != 'true' && github.event_name == 'pull_request'
+      run: |
+        # configure ecl-test.json with correct paths
+        jq --arg rootdir "${{ github.workspace }}" \
+          '.Regress.dropzonePath = $rootdir + "/install" + .Regress.dropzonePath | .Regress.setupExtraParams.OriginalTextFilesOsPath = $rootdir + "/install" + .Regress.setupExtraParams.OriginalTextFilesOsPath | .Regress.setupExtraDParams.HPCCBaseDir = $rootdir + "/install" + .Regress.setupExtraDParams.HPCCBaseDir | .Regress.regressionDir = $rootdir + "/regress" | .Regress.logDir = $rootdir + "/regress/log"' \
+          install/opt/HPCCSystems/testing/regress/ecl-test.json > ecl-test.json
+        mv -f ecl-test.json install/opt/HPCCSystems/testing/regress/ecl-test.json
+        # configure environment.xml to slavesPerNode=2, channelsPerNode=1
+        xmlstarlet ed -L -u 'Environment/Software/ThorCluster/@slavesPerNode' -v 2 -u 'Environment/Software/ThorCluster/@channelsPerSlave' -v 1 install/etc/HPCCSystems/environment.xml
```

Comment:
This option name is deprecated?

Reply:
no, at the moment, it's only manager/worker within the helm charts